### ISSUE TITLE
v-leave-from, v-leave-to css classes not triggering and before-leave javascript hook not triggering on page transition

### ISSUE THREAD

**Issue by geobotsar81** [inertiajs/inertia#644]
### Versions:

- `@inertiajs/inertia` version: #.#.# 0.8.6
- `@inertiajs/inertia-vue3` version: #.#.# 0.3.11
  "laravel/framework": "^8.12",
"inertiajs/inertia-laravel": "^0.4.1",

### Describe the problem:

I am having difficulties getting  the vue transition to work on page leave, and was wondering if this was a limitation when using vue3 with inertiajs. I have tried this with v-leave-to and v-leave-from in css classes and also by adding the before-leave javascript hook on the transition, and nothing seems to be triggering. v-enter-to and v-enter-from in css, and enter and after-enter Javascript Hooks are working ok

### Steps to reproduce:
Add vue transition element on inertiajs page element in Pages/Home/Index.vue

```
 <template>
    <transition name="page" @enter="enter" @after-enter="afterEnter" @before-leave="beforeLeave">
        <div v-if="animate">
            <main class="main">       
            </main>
        </div>
    </transition>
</template>
```

Added css classes

```
.page-enter-from {
    transition: all 0.6s ease-in;
    opacity: 0;
    transform: translateY(100px);
}

.page-enter-to {
    transition: all 0.6s ease-out;
    transform: translateY(0px);
    opacity: 1;
}

.page-leave-from {
    transition: all 0.6s ease-out;
    transform: translateY(0px);
    opacity: 1;
}

.page-leave-to {
    transition: all 0.6s ease-in;
    opacity: 0;
    transform: translateY(100px);
}
```

Leave transitions seems to not be triggering, only enter transitions.

Also in methods, the before leave hook seems to not be triggering when transitioning between pages:

```
 mounted() {
        this.animate = true;
    },
    data() {
        return {
            animate: false,
        };
    },
methods: {
        beforeLeave(el) {
            console.log("before leave");
        },
        beforeEnter(el) {
            console.log("before enter");
        },
        enter(el, done) {
            console.log("entered");
            done();
        },
        afterEnter(el) {
            console.log("after entered");
        },
        burgerIsClicked(menuIsOpen) {
            this.showMobileMenu = menuIsOpen;
        },
        footerLoaded() {
            //console.log("Footer is loaded");
            this.loading = false;
        },
    },
```

**Comment by claudiodekker** [comment]
Hey,

Inertia doesn't do anything specific / special / different in Vue 2 compared to Vue 3: In both cases, it's not Inertia that renders the Layouts and Pages, but rather it's Inertia telling Vue that an element has changed to a new one. This then causes Vue to not just update the element itself, but rather it causes Vue to re-compute what's changed on the entire page, and do everything that might be necessary related to that. This includes any transitions if they exist and are required.

Long story short: If this isn't working, it's with 99% certainty not an Inertia issue, but rather a Vue issue.

As such, I'm going to close this issue. However, if you feel absolutely certain that your issue still points out a bug in the library, please "@" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

Thanks!
